include/picongpu/plugins/saxs/Saxs.hpp

```diff
@@ -0,0 +1,815 @@
+                po::value< std::string >( & fileExtension )
+                    ->default_value( "bp" ),
+                "openPMD filename extension (this controls the backend "
+                "picked by the openPMD API)"
+            )(
+                (pluginPrefix + ".compression").c_str( ),
+                po::value< std::string >( & compressionMethod )
+                    ->default_value( "" ),
+                "Backend-specific openPMD compression method, e.g., zlib "
+                "(see `adios_config -m` for help)"
+            )
+            (
+                (pluginPrefix + ".memoryLayout").c_str( ),
+                po::value< std::string >( & memoryLayout )
+                    ->default_value( "mirror" ),
+                "Possible values: 'mirror' and 'split'"
+                "Output can be mirrored on all Host+Device pairs or"
+                " uniformly split over all nodes. Split can be used "
+                "when the output array is to big to store the complete "
+                "computed q-space on one device."
+            );
+        }
+
+
+        //! Get plugin name.
+        std::string pluginGetName( ) const override { return pluginName; }
+
+
+        //! Sets Mapping description for the saxs plugin.
+        void setMappingDescription( MappingDesc * cellDescription ) override
+        {
+            this->cellDescription = cellDescription;
+        }
+
+
+        void restart(
+            uint32_t timeStep,
+            const std::string restartDirectory
+        ) override
+        {
+            // TODO: Support for restarting.
+        }
+
+
+        void checkpoint(
+            uint32_t timeStep,
+            const std::string restartDirectory
+        ) override
+        {
+            // TODO: Support for restarting.
+        }
+
+
+private:
+
+
+        //! Prepare the plugin in the simulation initialization phase.
+        void pluginLoad( ) override
+        {
+            if ( !notifyPeriod.empty( ) )
+            {
+                /* Beam has to be initialized later as the domain sizes.
+                 *   The value retrieved by getDomainSize in
+                 *   CoordinateTransform.hpp is still set to (0,0,0) when the
+                 *   Saxs object is initialized.
+                 */
+                probingBeam = pmacc::memory::makeUnique< beam::SaxsProbingBeam
+                    >( );
+                // Set the steps at which the saxs amplitude is calculated.
+                Environment<>::get( ).PluginConnector( ).setNotificationPeriod(
+                    this,
+                    notifyPeriod
+                );
+                // Set the memory layout in use.
+                std::map< std::string, OutputMemoryLayout > layoutMap;
+                layoutMap["mirror"] = OutputMemoryLayout::MirrorOnNodes;
+                layoutMap["split"] = OutputMemoryLayout::SplitOverNodes;
+                outputLayout = layoutMap.at( memoryLayout );
```

Comment:
Just as a note, this will throw in case the user-set value for memory layout does not match one of the two supported values. Not sure if it would be easy to see the actual reason from the thrown message. In this regard, a binary yes/no switch as I suggested earlier is simpler. However, this is not a bug itself, of course.